cloudmanager/cvo_aws.go

```diff
@@ -57,7 +57,7 @@ type haParamsAWS struct {
 	MediatorKeyPairName       string   `structs:"mediatorKeyPairName,omitempty"`
 	PlatformSerialNumberNode1 string   `structs:"platformSerialNumberNode1,omitempty"`
 	PlatformSerialNumberNode2 string   `structs:"platformSerialNumberNode2,omitempty"`
-	MediatorAssignPublicIP    bool     `structs:"mediatorAssignPublicIP,omitempty"`
```

Comment:
would it not be enough to remove ',omitempty'  ?

why make it a pointer?

Reply:
Removing omitempty would cause the haparams.MediatorAssignPublicIP field to always be included in the JSON, even for non-ha scenarios. To fix that would require changing the haParams struct field to a ptr in the root object and not initialising the value for non-ha scenarios.

This would cause a much larger code change. Also, Boolean pointers were already being used on the root object (I assume for this exact reason).

Reply:
Makes sense.  I remember reading this some time ago.
https://stackoverflow.com/questions/37756236/json-golang-boolean-omitempty